the-collab-lab/tcl-10-smart-shopping-list
# Ly navbar

## Description

- added navbar with logo and title
- added copy and logout icon buttons that render conditionally if there is a token
- added copy function: onClick, a copy modal opens that shows the share token
- added logout function: onClick, removes the token from localStorage, state, and returns the user to the homepage

We wanted to enable users to easily navigate the site.  We also wanted users to be able to access multiple lists by being able to logout, and to make it easier to save tokens/share tokens with others.

## Related Issue
closes #54 


## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |
|  ✓ | :sparkles:Update styling     |

## Updates


### Before
<img width="1084" alt="Screen Shot 2020-09-23 at 10 51 49 PM" src="https://user-images.githubusercontent.com/50968795/94095871-a296ed80-fdf0-11ea-8a87-ecfbbbb2a8fa.png">

<img width="476" alt="Screen Shot 2020-09-23 at 10 51 26 PM" src="https://user-images.githubusercontent.com/50968795/94095753-65caf680-fdf0-11ea-8fab-cd0ba555a6d5.png">

### After

<img width="1063" alt="Screen Shot 2020-09-23 at 10 51 11 PM" src="https://user-images.githubusercontent.com/50968795/94095808-7d09e400-fdf0-11ea-8df4-09a57d493af8.png">
<img width="433" alt="Screen Shot 2020-09-23 at 10 50 46 PM" src="https://user-images.githubusercontent.com/50968795/94095816-7ed3a780-fdf0-11ea-9957-0affd4f79427.png">
<img width="453" alt="Screen Shot 2020-09-23 at 10 51 00 PM" src="https://user-images.githubusercontent.com/50968795/94095818-8004d480-fdf0-11ea-801d-c3e586319220.png">

## Testing Steps / QA Criteria

- Create a new list or use an old one. Check that onClick of the share icon, a copy modal opens. You can click copy and it will automatically copy to your clipboard without having to highlight the text.
- Once logged in with a token, click logout. Check that you are returned to the homepage, and that the two iconbutton disappear.
